content/200-concepts/100-components/01-prisma-schema/07-referential-actions.mdx

```diff
@@ -0,0 +1,327 @@
+
+```prisma file=schema.prisma highlight=5;add|4;delete
+model Post {
+    id        Int     @id @default(autoincrement())
+    title     String
+    author    User    @relation(fields: [authorId], references: [id])
+    author    User    @relation(fields: [authorId], references: [id], onDelete: SetNull)
+    authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `SetNull`?
+
+If a `User` is deleted then the corresponding `authorId` field in the `Post` table will be set to `Null`
+
+### SetDefault
+
+The `SetDefault` action will set relation [scalar fields](./data-model#scalar-fields) to their default values upon updated or deletion of a related record. If no defaults are provided for any of the scalar fields, a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=5;add|4;delete
+model Post {
+    id        Int     @id @default(autoincrement())
+    title     String
+    author    User    @relation(fields: [authorId], references: [id])
+    author    User    @relation(fields: [authorId], references: [id], onDelete: SetDefault)
+    authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `SetDefault`?
+
+If a `User` is deleted then the corresponding `authorId` field in the `Post` table will be set to its default value.
+
+
+## Upgrade guides
+
+The following lists a number of ways to upgrade depending on your scenario.
+
+1. Upgrading Prisma to 2.26.0 _**without**_ adding the feature flag.
+
+  As the preview feature is opt in, you should notice **no** difference to your database, its constraints or how it handles relations.
+
+2. Upgrading Prisma to 2.26.0 and _**enabling**_ the feature flag without adding custom referential actions. (Using the **new** defaults)
+
+  If you want to keep your current database constraints but upgrade you can enable the feature flag. To do so you must have a required relation in your schema, then run either a Prisma Migration or `db push`, this will change the default for `ON DELETE` from `CASCADE` TO `RESTRICT`.
+
+  The change is relevant for both the Prisma Client _and_ Migration. If an attempt is made to delete a referenced object, if their are related records, a database error will be thrown.
+
+3. Upgrading Prisma to 2.26.0, _**enabling**_ the feature flag and **adding custom referential actions**.
+
+  Once you have upgraded the Prisma version and added the feature flag to the generator block you can begin using the referential actions.
+
+  ### Using Introspection
+
+  You can begin by running an [Introspection](../introspection) on your database. This will update your data modals to use `ON DELETE CASCADE` and explicitly set `onDelete: Cascade` on your modals `@relation` fields. 
```

Comment:
```suggestion
  You can begin by running an [Introspection](../introspection) on your database. This will update your data models to use `ON DELETE CASCADE` and explicitly set `onDelete: Cascade` on your modals `@relation` fields. 
```